fc-community-contracts/src/FactChain1155.sol

```diff
@@ -55,12 +73,34 @@ contract FactChain1155 is Ownable, ERC1155, IFactChain1155 {
     function mint(uint256 id, uint256 value) external payable {
         if (msg.value != MINT_PRICE) revert BadMintPrice();
         if (value > MAX_TOKEN_SUPPLY) revert Greed();
+        if (value <= 0) revert ValueError();
+        if (supply[id] == SUPPLY_EXHAUSTED) revert SupplyExhausted();
+        if (supply[id] == 0) revert UnknownToken();
+
+        if (value > supply[id]) {
+            emit MintWithAdjustedValue(id, supply[id]);
+            _mint(msg.sender, id, supply[id], "");
+            supply[id] = SUPPLY_EXHAUSTED;
+        } else {
+            emit MintWithProvidedValue(id, value);
+            _mint(msg.sender, id, value, "");
+            supply[id] == value ? supply[id] = SUPPLY_EXHAUSTED : supply[id] -= value;
+        }
+    }
```

Comment:
Few comments:

- I think that the fact that `supply[id]` goes up is counter intuitive.
- By grouping errors `UnknownToken` and `SupplyExhausted`, we can simplify the code a bit
- I don't think we need to be helpful in the case where you try to mint to much. If value is above supply just fail.
- I think MINT_PRICE should be multiplied by the number of tokens you mint
- the `Greed` error is not needed

```solidity
        if (value <= 0) revert ValueError();
        if (supply[id] <= 0) revert SupplyExhausted();
        if (value > supply[id]) revert ValueError(); // Or some other error
        if (msg.value != value * MINT_PRICE) revert BadMintPrice();
        
        // I think it's better in this order
        supply[id] -= value;
        _mint(msg.sender, id, value, "");
        emit Mint(id, value);
```

Reply:
I've addressed all your comments except the first one.
If `supply` is a mapping to an int, then we need to cast the `value` to an int too before we can compare it to the supply, and AFAIR, casts are not safe with large values so I wanted to avoid any low-level trap.
To lower the impact on readability that this counter-intuitive limit could have, I've used a constant `SUPPLY_EXHAUSTED` 

While addressing the value > supply case, I think we don't want to fail but instead implement a refund logic and mint with the adjusted value. I've implemented this logic. Tell me wdyt.